collectors/modality-probe-debug-collector/src/lib.rs

```diff
@@ -399,6 +402,12 @@ impl Collector {
     /// Collect all new logs, return a report
     pub fn collect_report(&mut self) -> Result<Option<Report>, Error> {
         self.rbuf.clear();
+
+        // Insert paired wall clock time entry from previous report if needed
+        if let Some(e) = self.prev_paired_wall_clock_time.take() {
+            self.rbuf.push(e);
```

Comment:
I don't think this works correctly in the face of missing events.  If we discover in the next line that some entries were missed, then that means we've lost the appropriate-to-pair entry (and currently now associating this cached time entry with some random entry later on).

Reply:
I think in order to get this right, you may have to write a test that creates conditions wherein the paired content gets overwritten, then we correctly discover this fact on the next read and decide to drop the cached wall-clock-time (WCT) entry rather than pairing it spuriously.

Reply:
An alternative approach when we detect that the paired content has been missed would be to downgrade the paired WCT into an unpaired WCT entry and keep it around.

Reply:
Good catch, will revisit this.